archiver.go

```diff
@@ -310,6 +310,7 @@ func writeNewSymbolicLink(fpath string, target string) error {
 	if err != nil {
 		return fmt.Errorf("%s: making directory for file: %v", fpath, err)
 	}
+	os.Remove(fpath)
```

Comment:
We should only remove it if it's a symbolic link, right? Not safe to overwrite existing files with actual content.

Also might need to honor the various OverwriteExisting parameters for each archiver, for example, zip's: https://github.com/mholt/archiver/blob/d8fc61af55e91ce2231ac9e1965cb9e77b435be8/zip.go#L25

Reply:
@mholt Please consider new changes. 
Now `zip` and `tar` archives which currently use `writeNewSymbolicLink` method, use their `OverwriteExisting` to remove existing symbolic links safely.